### DISCUSSION TITLE
Multiple slices configuration

### DISCUSSION THREAD

**Discussion by jamjad333** [srsran/srsRAN_Project#602] [General Help]
Hi everyone,thanks for your work!
I am facing issue regarding slices as i am trying to connect through multiple slices.Does srsRAN support multiple slices simultaneously?
I have following setup:
Core: open5gs v2.4
RAN: srsRAN/USRP B210
UE: Quectel RM500Q-AE

As you can see i have multiple slices configured on my quectel module and i am able to connect APN's only which refers to sst=1 in my core but no response when i try to connect APN with sst value other than '1'.

![Screenshot from 2024-05-06 13-44-03](https://github.com/srsran/srsRAN_Project/assets/159905841/b4f17208-e01e-443c-8456-9908953134c0)

![Screenshot from 2024-05-06 14-47-00](https://github.com/srsran/srsRAN_Project/assets/159905841/b36cf321-5124-4920-af79-df54b6febade)

![Screenshot from 2024-05-06 14-47-00](https://github.com/srsran/srsRAN_Project/assets/159905841/b6192fbc-8ee6-4169-ba9a-bb017f997c8c)

Core side:

![Screenshot from 2024-05-06 14-46-02](https://github.com/srsran/srsRAN_Project/assets/159905841/328a16bf-2169-4289-bf47-0c8581350170)

gnb logs attached:
[gnb_log.zip](https://github.com/srsran/srsRAN_Project/files/15220864/gnb_log.zip)

**Comment by improperaffo** [comment]
Hi @jamjad333, could you share your `gnb.conf` file?

Also, Ubuntu has a built in modem manager section in the settings that might make it easier to manage things like APNs when using Quectel modems. Checkout this screenshot: 
![image](https://github.com/srsran/srsRAN_Project/assets/96285590/cd665a13-7ddb-4a5f-bc6d-93ea02086eba)

**Comment by jamjad333** [comment]
Thanks for the feedback,here is the config file

[config_GNB.zip](https://github.com/srsran/srsRAN_Project/files/15233883/config_GNB.zip)


Even if i try to i connect with second APN it automatically connects to one configured with sst=1

**Reply by improperaffo** [reply]
That is weird, we usually have a line from the AMF displaying the NSSAI of the UE:
```
05/07 09:49:59.443: [gmm] INFO: UE SUPI[imsi-001010123456743] DNN[internet] S_NSSAI[SST:3 SD:0x0] (../src/amf/gmm-handler.c:1236)

```

What are the NSSAI values that are set in the core and in the UE database? Also, we have noticed that sometimes, if you leave the SD field empty, it gets set to 0xffffff, sometimes to 0x000000 (different projects have different implementations), so I would not leave it blank.

This is how we are setting our slices in the `gnb.conf`:
```
slicing:
 - sst: 1
   sd: 0
 - sst: 1
   sd: 1
 - sst: 2
   sd: 1
 - sst: 3
   sd: 0
```

**Reply by jamjad333** [reply]
in core side:
sst=1
sst=2
with no sd value

in DB:

![Screenshot from 2024-05-07 12-28-38](https://github.com/srsran/srsRAN_Project/assets/159905841/3c6a391b-37be-438a-9d41-ca15373ae0b8)

and in gnb as i shared with you already and i do have NSSAI info in AMF as below:

![Screenshot from 2024-05-07 12-31-38](https://github.com/srsran/srsRAN_Project/assets/159905841/a5747399-9bac-4814-818b-c968f493729e)

**Reply by jamjad333** [reply]
furthermore when i define NSSAI with sd info,it gives me following errors:

![Screenshot from 2024-05-07 12-47-45](https://github.com/srsran/srsRAN_Project/assets/159905841/1772411f-c31c-4bb1-9806-46054aea11f0)

**Reply by improperaffo** [reply]
After you define the core NSSAI with the SD value, you should also match that in the UE database and in the gNB configuration.
Sometimes the UEs keep some context about the SIM card, so it is good to remove it and / or reboot the modem.

Incidentally, could I ask you to run `ping 8.8.8.8 -I wwan0 -i 0.02` when the modem is connected (even without slicing)? This is not related to the slices, but I also own a Quectel modem, and it reboots every time I try to generate some mildly intense traffic, so I wanted to know if that is just my problem or if this happens to everybody. Thank you

**Reply by jamjad333** [reply]
so it means the UE can connects to two seperate APNs normally,there's no limitations related to srsRAN,right?

Regarding ping it does not give me any such issue.

**Reply by improperaffo** [reply]
I haven't tried different APNs, I just tried different slices. And i can confirm that a UE can change slice, for example if the core moves the UE to a different slice, or if a controller (like the RIC) moves the UE to a different slice.

Regarding the ping, can you also run `speedtest-cli` over the interface created by the modem? Are you using external power for the modem?

**Reply by jamjad333** [reply]
No i am not using any external power rather than connection cable
For ping i get normal response and you need to check pcap because in terminal it will show you complete loss but in pcap you will see normal requests and responses even with intense traffic,attached below

[quectel_pcap.zip](https://github.com/srsran/srsRAN_Project/files/15237023/quectel_pcap.zip)

**Reply by Alan1321** [reply]
@improperaffo 

Were you able to establish multiple PDU sessions using srsRAN with the Quectel modem?
If so, were you able to create additional PDU sessions after the initial registration, during runtime?

Also, when creating PDU sessions during runtime, were you able to select between multiple statically configured slices (e.g., different S-NSSAIs), or was the slice fixed at registration time?

You mentioned earlier: “I also own a Quectel modem, and it reboots every time I try to generate some mildly intense traffic.”
I ran into the same issue, and in my case it was power-related. The modem requires external power; powering it solely through USB from the host device is not sufficient under higher traffic loads.

**Reply by improperaffo** [reply]
We were only able to establish a single PDU session per modem. We were able to move UEs among slices, using some control messages from the gNB side (an RRC reconfiguration if I recall correctly).

Thanks for the tip with the modem

**Comment by jamjad333** [comment]
> I haven't tried different APNs, I just tried different slices. And i can confirm that a UE can change slice, for example if the core moves the UE to a different slice, or if a controller (like the RIC) moves the UE to a different slice.
> 
> Regarding the ping, can you also run `speedtest-cli` over the interface created by the modem? Are you using external power for the modem?

hi,can you tell me how you tested 'move UE to different slice by core or controller(like RIC)'

**Reply by improperaffo** [reply]
Hi. We have developed some simple logic that send commands to the gNB over ZMQ, which trigger `du_ue_ric_configuration_procedure::handle_mac_ue_reconfig()` in `lib/du_manager/procedures/du_ue_ric_configuration_procedure.cpp`

**Comment by pgawlowicz** [comment]
@jamjad333 Please check this discussion: https://github.com/srsran/srsRAN_Project/discussions/756#discussioncomment-10240374
